EIPS/eip-hard-gas-target.md

```diff
@@ -0,0 +1,60 @@
+---
+eip: xxxx
+title: Hardcoded Block Gas Target
+author: Philippe Castonguay (@PhABC)
+discussions-to: xxxx
+status: Draft
+type: Standards Track
+category: Core
+created: 2020-03-13
+
+---
+
+## Simple Summary
+
+Hardcode the block gas target (previously known as *block gas limit* before [EIP-1559](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md)) to `12,500,000` gas per block.
+
+## Abstract
+
+Ethereum's block gas target is currently dictated by block producers and is not enforced when validating blocks in clients. This EIP proposes to hardcode the block gas target to `12,500,000`.
+
+## Motivation
+
+Both Ethereum's Proof of Work and Proof of Stake designs assume that block producers are financially rational, but does not assume block producers to be benevolent. There is one exception however, and it is when block producers choose the gas targe of a block (previously known as block gas limit), where it is assumed that block producers care about the long term health and decentralisation of the chain. Indeed, the block gas target is one of the only parameters in Ethereum that is not dictated by node consensus, but instead is chosen by block producers. This decision was initially made to allow urgent changes in the block gas target if necessary. Both drastically increasing or decreasing this parameter could have serious consequences that may not be desired. It is therefore a critical parameter that should require node consensus to avoid any sudden harmful change imposed by a small number of actors on the rest of the network.
+
+## Specification
+This EIP assumes that EIP-1559 is implemented and blocks' `gasLimit` was replaced with `gasTarget`.
+
+#### Block Header
+
+As of `FORK_BLOCK_NUMBER`, remove `gas_target` field from block headers.
+
+#### Consensus
```

Comment:
This is already implied by 1559.

Reply:
Is it? I believe gas_target is still part of block headers with EIP-1559.

Reply:
I'm referring to the consensus section here. If the gas limit is required to be 12.5mm per consensus then the gas used will be appropriately constrained pre and post-1559.

Reply:
Ah, I thought your original comment was referring to `As of `FORK_BLOCK_NUMBER`, remove `gas_target` field from block headers.`

However, EIP-1559 does not imply the hardcoded gasLimit is part of the consensus. Current EIP-1559 spec uses the gas_target from a block's header, not this proposed hardcoded constant. As per previous comments you and Micah made, the gas_target provided in headers will remain, so it's important that the consensus algorithm implementations are changed to using the hard-coded gasLimit, not the header's gas limit.

Reply:
Ah yes, if the limit was removed from the header it would need to be specified. Since the limit will stay in the header, why not a consensus rule that requires `block.gas_limit == 12_500_000`? The rules for EIP-1559 should stay the same in that case?

Reply:
Good point, i'll remove the redundant stuff.